### ISSUE TITLE
Extend the plugins API to enable requesting a plugin icon

### ISSUE THREAD

**Issue by absoludity** [kubeapps/kubeapps#3748]
### Description:

We currently have hard-coded switch statement in the dashboard to return the icon for a plugin.

This could be a nice extension to the plugin API, serving (or returning) the icon for a plugin, if we can ensure it's cached correctly.

**Comment by antgamdia** [comment]
Good idea, mid/long term we might also want to think of UI components being served by the plugin if the non-core functionality happens to diverge considerably.  
I do like the idea, but, as you mention, I'm not sure atm how to leverage from the webpack process for compressing and generating the final bundle if we have to request the plugin img each time. But sure it is worths having a look at it.
